### PR TITLE
Upgrade nixpkgs

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ci/build.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
   timeoutInMinutes: 360
   pool:
     name: 'ubuntu_20_04'
-    demands: assignment -equals default
+    demands: assignment -equals nixpkgs-upgrade
```
Comment (cocreature) [inline ci/build.yml:39]: I’ll remove that before merging.

File: nix/bazel.nix [hunk 2]
```diff
@@ -29,7 +29,20 @@ let shared = rec {
     zip
     ;
 
-  scala = pkgs.scala_2_12;
+    scala = (pkgs.scala_2_12.override { }).overrideAttrs (attrs: {
+      # Something appears to be broken in nixpkgs' fixpoint which results in the
```
[the remark this answers is not in the document]
Comment (aherrmann-da) [reply nix/bazel.nix:33]: Is there an issue on nixpkgs about it?

File: nix/nixpkgs.nix [hunk 3]
```diff
@@ -46,12 +46,34 @@ let
...
       src = pkgs.fetchurl {
         url = "https://www.scala-lang.org/files/archive/${name}.tgz";
         sha256 = "3520cd1f3c9efff62baee75f32e52d1e5dc120be2ccf340649e470e48f527e2b";
       };
     });
+   haskell = pkgs.haskell // {
+     packages = pkgs.haskell.packages // {
+       integer-simple = pkgs.haskell.packages.integer-simple // {
+        ghc8103 = pkgs.haskell.packages.integer-simple.ghc8103.override {
+          ghc = pkgs.haskell.compiler.integer-simple.ghc8103.overrideAttrs (old: {
```
Comment (aherrmann-da) [inline nix/nixpkgs.nix:60]: Why is this needed?

**Review by aherrmann-da — APPROVED** [review]
Thank you! LGTM

File: nix/nixpkgs.nix (hunk 3 above)
Comment (cocreature) [reply nix/nixpkgs.nix:60]: Otherwise GHC fails to find `otool`. Added a comment (I already had, just forgot to push it :facepalm:)

File: nix/bazel.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (cocreature) [reply nix/bazel.nix:33]: not yet, I was planning to look into it a bit more but haven’t found the time.
